qiskit/quantum_info/operators/dihedral/dihedral.py

```diff
@@ -19,7 +19,7 @@
 from qiskit.exceptions import QiskitError
 from qiskit.quantum_info.operators.base_operator import BaseOperator
 from qiskit.quantum_info.operators.operator import Operator
-from qiskit.quantum_info.operators.pauli import Pauli
```

Comment:
I suspect that this file probably isn't the only extant code that will import `Pauli` from this location.  It might be worthwhile keeping the `qiskit/quantum_info/operators/pauli.py` file, never importing it within Terra, and having that file just expose `Pauli` with a deprecation warning saying "import this from `qiskit.quantum_info` directly".  We could leave that up for a deprecation cycle, and then remove it.

The `pauli_group` removals can definitely happen now, though.